### DISCUSSION TITLE
Can't allocate memory error on running train_test_split_edges.py

### DISCUSSION THREAD

**Discussion by adg-git** [pyg-team/pytorch_geometric#2747] [Q&A]
I am getting the following error 
RuntimeError: [enforce fail at CPUAllocator.cpp:65] . DefaultCPUAllocator: can't allocate memory: you tried to allocate 141957139984 bytes. Error code 12 (Cannot allocate memory)

for line 
https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/utils/train_test_split_edges.py#L73

I have 380K nodes in my graph.

Has anybody experienced similar issues? Any suggestions on how to resolve it would be greatly appreciated.

**Comment by rusty1s** [comment]
The current `train_test_split_edges` method does not scale well to large graphs, since it will create a dense `BoolTensor` for negative sampling. An alternative is to create negative samples via `torch.utils.negative_sampling(method='sparse')`.
